src/transforms/push.ts

```diff
@@ -133,49 +110,47 @@ export const processPush = async (github: GitHubAPI, payload) => {
 			log
 		);
 
-		const commits = await Promise.all(
-			shas.map(async (sha) => {
-				const {
-					data,
-					data: { commit: githubCommit },
-				} = await github.repos.getCommit({
-					owner: owner.login,
-					repo,
-					ref: sha.id,
-				});
-
-				const { files, author, parents, sha: commitSha, html_url } = data;
-
-				const { author: githubCommitAuthor, message } = githubCommit;
-
-				// Jira only accepts a max of 10 files for each commit, so don't send all of them
-				const filesToSend = files.slice(0, 10);
-
-				// merge commits will have 2 or more parents, depending how many are in the sequence
-				const isMergeCommit = parents?.length > 1;
-
-				return {
-					hash: commitSha,
-					message,
-					author: getJiraAuthor(author, githubCommitAuthor),
-					authorTimestamp: githubCommitAuthor.date,
-					displayId: commitSha.substring(0, 6),
-					fileCount: files.length, // Send the total count for all files
-					files: filesToSend.map((file) =>
-						mapFile(file, repo, owner.name, sha.id)
-					),
-					id: commitSha,
-					issueKeys: sha.issueKeys,
-					url: html_url,
-					updateSequenceId: Date.now(),
-					flags: isMergeCommit ? ["MERGE_COMMIT"] : undefined,
-				};
-			})
-		);
+		const commits: JiraCommit[] = (await getSpecificGithubCommits(github, {
+			commitRefs: shas.map(s => s.id),
+			owner: owner.login,
+			repoName
+		})).map((commit: GithubCommit): JiraCommit => {
+			const {/* files,*/
+				oid,
+				abbreviatedOid,
+				author,
+				authoredDate,
+				parents: { totalCount: parentCount },
+				url,
+				message,
+				changedFiles,
+			} = commit;
+
+			// Jira only accepts a max of 10 files for each commit, so don't send all of them
+			// const filesToSend = files.slice(0, 10);
+
+			// merge commits will have 2 or more parents, depending how many are in the sequence
+			const isMergeCommit = parentCount !== 0;
```

Comment:
I think this is the issue. All commits have 1 parent (main/master). So any commit that is made is flagged as a merge commit.